src/main/java/io/neonbee/internal/verticle/ServerVerticle.java

```diff
@@ -137,27 +95,78 @@ public void stop(Promise<Void> stopPromise) throws Exception {
         }) : succeededFuture().<Void>mapEmpty()).onComplete(stopPromise);
     }
 
+    private Future<Router> createRouter(ServerConfig config) {
+        // the main router of the server verticle
+        Router router = Router.router(vertx);
+
+        // instead of creating new routes, vert.x recommends to add multiple handlers to one route instead. to prevent
+        // sequence issues, block scope the variable to prevent using it after the endpoints have been mounted
+        Route rootRoute = router.route();
+
+        return createErrorHandler(config.getErrorHandlerClassName(), config.getErrorHandlerTemplate(), vertx)
+                .compose(errorHandler -> {
+                    rootRoute.failureHandler(errorHandler);
+                    rootRoute.handler(new LoggerHandler());
+                    rootRoute.handler(BodyHandler.create(false /* do not handle file uploads */));
+                    rootRoute.handler(new CorrelationIdHandler(config.getCorrelationStrategy()));
+                    rootRoute.handler(TimeoutHandler.create(SECONDS.toMillis(config.getTimeout()),
+                            config.getTimeoutStatusCode()));
+                    rootRoute.handler(new CacheControlHandler());
+                    rootRoute.handler(new InstanceInfoHandler());
+
+                    createSessionStore(vertx, config.getSessionHandling()).map(SessionHandler::create)
+                            .ifPresent(sessionHandler -> rootRoute
+                                    .handler(sessionHandler.setSessionCookieName(config.getSessionCookieName())));
+
+                    return succeededFuture(router);
+                }).onFailure(e -> {
+                    LOGGER.error("Server could not be started", e);
```

Comment:
This is a wrong error message, the thing that fail here, is the error handler which could not be created and it should be checked with .onFailure only with the error handler.